### PR TITLE
language-chatgpt adaptor

### PR DESCRIPTION
**Pull request by haftamuk**
## Summary

language-chatgpt adaptor added

## Details

A promptChatGpt(prompt) function is created to interact with OpenAI gpt-3.5-turbo model.

## Issues

[API Reference](https://platform.openai.com/docs/libraries/node-js-library)

## Review Checklist

Before merging, the reviewer should check the following items:

- [ ] Does the PR do what it claims to do?
- [ ] If this is a new adaptor, has the migration tool been run and the
      migration guide followed? 
- [ ] Are there any unit tests? Should there be?
- [ ] Is there a changeset associated with this PR? Should there be? Note that
      dev only changes don't need a changeset.

### CODE REVIEW

File: packages/chatgpt/assets/rect-512x200.png
Comment (mtuchi) [inline packages/chatgpt/assets/rect-512x200.png:1]: @haftamuk please update this to chat gpt 521x200.png

File: packages/chatgpt/src/Adaptor.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import {
+  execute as commonExecute,
+  composeNextState,
+} from '@openfn/language-common';
+import { expandReferences } from '@openfn/language-common/util';
+import { request } from './Utils';
+import 'dotenv/config';
```
Comment (mtuchi) [inline packages/chatgpt/src/Adaptor.js:7]: We use `state.configuration` for setting up credentials, please remove this library

File: packages/chatgpt/configuration-schema.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "properties": {
+    "OPENAI_SECRET_KEY": {
+      "title": "OPENAI_SECRET_KEY",
```
Comment (mtuchi) [inline packages/chatgpt/configuration-schema.json:5]: the title should be `apiKey`

File: packages/chatgpt/src/Adaptor.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+/**
+ * Prompt OpenAI model.
+ * @public
+ * @example
+ * promptChatGpt("prompt")(state)
+ * @function
+ * @param {string} prompt - User textual prompt
+ * @returns {Operation}
+ */
+export function promptChatGpt(prompt) {
```
Comment (mtuchi) [inline packages/chatgpt/src/Adaptor.js:20]: A slight suggestion on the function signature that will give users more flexibility 
```
@param {string} prompt - User textual prompt
@param {object} options - Options for textual prompt eg model, role
promptChatGpt(prompt, options)
```

File: packages/chatgpt/CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
+# @openfn/language-chatgpt
+
+## 3.0.1
```
Comment (josephjclark) [inline packages/chatgpt/CHANGELOG.md:3]: The changelog probably needs updating  :)

**Comment by josephjclark** [comment]
Hi @haftamuk - thank you for this contribution!

We're very keen to get it merged in. Are you able to respond to the feedback this week? If not, do you mind if we take it over?

**Comment by haftamuk** [comment]
> Hi @haftamuk - thank you for this contribution!
> 
> We're very keen to get it merged in. Are you able to respond to the feedback this week? If not, do you mind if we take it over?

Hey.
I will spend enough time to address the feedbacks this week. Also Please feel free to dive in on any other comments. I will be pleased to see the work getting merged too.

**Comment by josephjclark** [comment]
Thank you @haftamuk! We'll give you a few days and maybe take this over next week. We'll let you know if we start working on anything so that we're not duplicating effort.

**Review by josephjclark — COMMENTED** [review]
Whoops, just realised I had two review comments still pending - posting those now!

**Comment by josephjclark** [comment]
Hi @haftamuk - we'd like to merge this, make a few adjustments and put out a release. Is that OK with you?

**Comment by josephjclark** [comment]
Hi @haftamuk,

I am really sorry but I'm going to close this PR for now.

There was a minute there when we were going to take it over and release it ourselves. But we missed that boat and we've got other priorities at the moment.

I'd welcome a new PR which addresses the comments above and updates the library version. But until then the PR must be closed :pray:

**Pull request closed without merging**